### ISSUE TITLE
bind reusable component to different data

### ISSUE THREAD

**Issue by Maikel1963** [vizia/vizia#439]
Hello
I am impressed by the low-level description in your vizia book. For me (who hasn't make GUI before) it was exactly the level I needed to understand what I am  doing. But one point hasn't been described clearly.
"add multiple counters all bound to the same data (or different data)" 
How do I bind the different components to different data. 
I started with 3 counters in struct AppData (count1, count2 and count3) but I didn't manage to retrieve the information which of the e.g. Increment buttons had been pressed.

**Comment by geom3trik** [comment]
You are quite right, the example in the book wasn't properly set up to be completely reusable. I've just updated the book with guidance on how to achieve this. The gist is that the counter should provide callbacks which the user can use to emit their own events when the buttons are pressed. E.g.

```rust
Counter::new(AppData::count)
    .on_increment(|cx| cx.emit(AppEvent::Increment))
    .on_decrement(|cx| cx.emit(AppEvent::Decrement));
```
The user can then handle separate events for separate pieces of data, or one event with an index, etc.

To do this is a little involved, requiring custom modifiers and internal event handling, but I've updated the book to reflect this. Let me know if anything still isn't clear.

**Comment by Maikel1963** [comment]
Thanks for the fast answer.
I am working through the book again to not miss anything. 
I encountered a problem which I do not remember to have it before. In chapter 12 - "Lenses and Data Binding" there is a code snippet given:
`#[derive(Lens)]`
`pub struct AppData {`
`    count: Count,`
`}`
This does not compile if you do not replace "Count" by "i32"
I'll write again when I am finished 
;-)
